src/main/java/com/gregtechceu/gtceu/api/capability/recipe/ItemRecipeCapability.java

```diff
@@ -339,6 +331,57 @@ public int getMaxParallelRatio(IRecipeCapabilityHolder holder, GTRecipe recipe,
         return minMultiplier;
     }
 
+    private Object2IntMap<ItemStack> getIngredientStacks(IRecipeCapabilityHolder holder) {
+        Object2IntMap<ItemStack> map = new Object2IntOpenHashMap<>();
+        Object2IntMap<ItemStack> result = new Object2IntOpenHashMap<>();
```

Comment:
Just use `Object2IntOpenCustomHashMap` with the correct hash strategy to not have to loop all entries for a `Map#containsKey`